lib/spotlight/dor/indexer.rb

```diff
@@ -83,6 +91,31 @@ def add_point_bbox(sdb, solr_doc)
       def add_series(sdb, solr_doc)
         solr_doc['series_ssi'] = sdb.smods_rec.series
       end
+
+      # @return [Array{String}] The IDs from geonames //subject/geographic URIs, if any
+      def extract_geonames_ids(sdb)
+        sdb.smods_rec.subject.map do |z|
+          uri = z.geographic.attr('valueURI')
+          m = %r{^https?:\/\/sws\.geonames\.org\/(\d+)}i.match(uri.nil? ? '' : uri.value)
+          m ? m[1] : nil
+        end.reject(&:nil?).reject(&:empty?)
+      end
+
+      # Fetch remote geonames metadata and format it for Solr
+      # @param [String] id geonames identifier
+      # @return [String] Solr WKT/CQL ENVELOPE based on //geoname/bbox
+      def get_geonames_api_envelope(id)
+        url = "http://api.geonames.org/get?geonameId=#{id}&username=#{Spotlight::Dor::Resources::Engine.config.geonames_username}"
+        xml = Nokogiri::XML Faraday.get(url).body
+        bbox = xml.at_xpath('//geoname/bbox')
+        return if bbox.nil?
+        min_x, max_x = [bbox.at_xpath('west').text.to_f, bbox.at_xpath('east').text.to_f].minmax
+        min_y, max_y = [bbox.at_xpath('north').text.to_f, bbox.at_xpath('south').text.to_f].minmax
+        "ENVELOPE(#{min_x},#{max_x},#{max_y},#{min_y})"
+      rescue RuntimeError => e
+        logger.warn("Error fetching/parsing #{url} -- #{e.message}")
```

Comment:
I expect that you will write the tests for the code that you write/change. I don't really care about `get_file_content`.